samples/Hardware.Esp32.Rmt/InfraredRemoteReceiver/InfraredRemoteReceiverSample/Decoder.cs

```diff
@@ -0,0 +1,157 @@
+
+            var message = ExtractRawPayload(receivedSignal);
+
+            if (isRepeat)
+            {
+                return lastData;
+            }
+
+            if (isHeaderMark && isSpaceMark && (receivedSignal.Length == SignalLength) && isEndOfTransmission)
+            {
+                signalData.Protocol = Protocol;
+                var address = ExtractAddress(message);
+                var command = ExtractCommand(message);
+                signalData.RawAddress = address;
+                signalData.RawCommand = command;
+                signalData.Payload = message;
+                if (UseLessSignificantBitFirst)
+                {
+                    signalData.AddressNumber = Convert.ToInt32(Reverse(address), 2);
+                    signalData.CommandNumber = ToInt32(Reverse(command));
+                    var temp = Convert.ToInt32(Reverse(command), 2);
+                }
+                else
+                {
+                    signalData.AddressNumber = ToInt32(address);
+                    signalData.CommandNumber = ToInt32(command);
+                }
+
+                lastData = signalData;
+            }
+
+            return signalData;
+        }
+
+        protected virtual string ExtractCommand(string message)
+        {
+            return message.Substring(AddressBits, CommandBits);
+        }
+
+        protected virtual string ExtractAddress(string message)
+        {
+            return message.Substring(0, AddressBits);
+        }
+
+        private string ExtractRawPayload(RmtCommand[] response)
+        {
+            StringBuilder sb = new StringBuilder();
+            for (int i = 1; i < response.Length - 1; i++)
+            {
+                var rmtCommand = response[i];
+                if (Match(rmtCommand.Duration1, OneSpace))
+                {
+                    sb.Append("1");
+                }
+                else
+                {
+                    sb.Append("0");
+                }
+            }
+
+            return sb.ToString();
+        }
+
+        protected bool Match(int tick, int expectedTick)
+        {
+            var ticksLow = expectedTick * (1 - SignalLengthTolerance);
+            var ticksHigh = expectedTick * (1 + SignalLengthTolerance);
+            return (tick >= ticksLow &&
+                    tick <= ticksHigh);
+        }
+
+        private string Reverse(string valueToBeReversed)
+        {
+            StringBuilder sb = new StringBuilder();
+            for (int i = valueToBeReversed.Length - 1; i >= 0; i--)
+            {
+                sb.Append(valueToBeReversed[i]);
+            }
+
+            return sb.ToString();
+        }
+
+        private int ToInt32(string binaryValue)
+        {
+            int dec_value = 0;
+
+            int base1 = 1;
+
+            for (int i = binaryValue.Length - 1; i >= 0; i--)
+            {
+                if (binaryValue[i] == '1')
+                    dec_value += base1;
```

Comment:
We're using explicit brackets
```suggestion
                {
                    dec_value += base1;
                }
```